js/main.js

```diff
@@ -70,31 +70,35 @@ function onDOMLoad(event) {
     var entryRendered = renderEntry(data.entries[entryNum]);
     $entryList.appendChild(entryRendered);
   }
-  if (data.view === 'entries') {
-    $formDiv.className = data.view + ' hidden';
-    data.view = 'entries';
-    $entriesDiv.className = data.view;
+  for (var i = 0; i < $views.length; i++) {
+    if ($views[i].getAttribute('data-view') === data.view) {
+      $views[i].className = 'view';
+    } else {
+      $views[i].className = 'view hidden';
+    }
```

Comment:
Once you make the revision I mentioned below, you should be able to use your `switchViews` here